exp-player/addon/mixins/warn-on-exit-controller.js

```diff
@@ -0,0 +1,33 @@
+import Ember from 'ember';
+
+export default Ember.Mixin.create(Ember.Evented, {
+    showWarning: false,
+    forceExit: false,
+
+    session: Ember.computed(function() {
+        return this.get('model.session') || this.get('model');
```

Comment:
That seems more like an issue with a mixin trying to fit into more than one use case, honestly. 

It might even be a good idea to specify the path to the session explicitly- if the mixin can't find the model it expects, guessing blindly might end with it checking the wrong model by accident.

(I agree that hashes get weird when the model hook is invoked in different ways, but that's a different issue)